app/models/ability.rb

```diff
@@ -4,101 +4,94 @@ class Ability
-          can :manage, :dossiers_opal
+    end
+
+    if agent_or_user.is_a? Agent
+      agent = agent_or_user
+
+      can :index, Projet
+
+      return if projet.blank?
+      return unless is_agent_of_projet?(agent, projet)
+
+      return can :manage, :all if agent.admin?
+      return can :read,   :all if agent.siege?
+
+      if agent.pris?
+        if projet.statut.to_sym == :prospect
+          can :read,                   Projet
+          can :recommander_operateurs, Projet
         end
+      end
 
-      elsif agent_or_user.operateur? && project_belongs_to_operateur(projet, agent_or_user)
+      if agent.instructeur?
+        can :create, :dossiers_opal if projet.statut.to_sym == :transmis_pour_instruction
+        can :read,   Projet         if projet.status_already :transmis_pour_instruction
+      end
+
+      if agent.operateur?
         if projet.statut.to_sym == :prospect
           can :read, Projet
-        elsif projet.statut.to_sym != :prospect && projet.status_not_yet(:transmis_pour_instruction)
+        elsif projet.status_not_yet(:transmis_pour_instruction)
           can :manage, AvisImposition
           can :manage, Demande
           can :manage, :demandeur
           can :manage, Occupant
           can :manage, Projet
-        elsif projet.statut.to_sym != :en_cours && projet.statut.to_sym != :prospect
+        else
           can :read, Projet
         end
       end
     end
 
-    if agent_or_user.is_a?(User)
-      return if projet.blank?
-      if projet.locked_at.nil?
-        can :manage, AvisImposition
-        can :manage, Demande
-        can :manage, :demandeur
-        can :manage, :eligibility
-        can :manage, Occupant
-        can :manage, Projet
-      elsif project_belongs_to_demandeur(projet, agent_or_user)
-        can :read, :eligibility
-        can :read, Projet
-      end
-    end
+    define_payment_registry_abilities(agent_or_user, projet)
+    define_payment_abilities(agent_or_user, projet)
+    define_document_abilities(agent_or_user, projet)
   end
 
 private
 
   def define_payment_registry_abilities(agent_or_user, projet)
-    if projet.present?
-      if projet.payment_registry.blank?
-        project_transmited = !projet.status_not_yet(:transmis_pour_instruction)
+    can :read, PaymentRegistry, projet_id: projet.id
 
-        can :create, PaymentRegistry if agent_or_user.try(:operateur?) && project_transmited
-      else
-        can :read,   PaymentRegistry, projet_id: projet.id
-      end
+    if agent_or_user.try(:operateur?) && projet.status_already(:transmis_pour_instruction) && projet.payment_registry.blank?
+      can :create, PaymentRegistry
     end
   end
 
   def define_payment_abilities(agent_or_user, projet)
-    alias_action :new,  :create, to: :add
-    alias_action :edit, :update, to: :modify
-    alias_action :edit, :update, :destroy, to: :modify_destroy
     if projet.payment_registry.present?
```

Comment:
Tu m'avais initialement parlé de rajouter des vérifs "if projet belongs to agent" tu ne les as pas mis ici ?